pkg/skaffold/deploy/resource/deployment.go

```diff
@@ -79,3 +88,9 @@ func (d *Deployment) WithStatus(details string, err error) *Deployment {
 	d.UpdateStatus(details, err)
 	return d
 }
+
+func (d *Deployment) WithDone(details string, err error) *Deployment {
```

Comment:
Is this function only used for unit tests?

If so, maybe we could move it to the test file as a helper function --


```go
func WithDone(d *Deployment, details string, err error) *Deployment
```

Reply:
sounds good.